### ISSUE TITLE
Type annotation which is dynamic variable

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3511]
### Discussed in https://github.com/microsoft/pyright/discussions/3510

<div type='discussions-op-text'>

<sup>Originally posted by **amacfie** May 25, 2022</sup>
```python
for t in [str, float]:
    def f(x: str) -> t:
        return t(x) + 1
    print(f("3.14"))
```
Pyright finds no errors in this. Is there different type syntax I should use or is this outside of Pyright's scope?</div>

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.250, which I just published. It will also be included in the next release of pylance.
